mypy/test/data.py

```diff
@@ -53,11 +53,11 @@ def parse_test_cases(
             while i < len(p) and p[i].id != 'case':
                 if p[i].id == 'file':
                     # Record an extra file needed for the test case.
-                    files.append((os.path.join(base_path, p[i].arg),
+                    files.append((os.path.join(base_path, p[i].arg or ''),
```

Comment:
Hm, I rewrote it using the assert, then realized that arg may be None if someone puts `[file]` in a testcase. That's nonsense. In general the parsing of test cases is pretty crummy, so I don't consider it my job to fix it.